### PR TITLE
Add first question to survey

### PR DESCRIPTION
**Pull request by katrinacrisostomo**
Issue: https://github.com/uwblueprint/somo/issues/45

Please don't merge until Bree and Nick take a look as I wasn't able to finish it with them

### CODE REVIEW

File: app/models/survey.rb [hunk 1]
```diff
@@ -7,11 +7,14 @@
 #  description                          :text
 #  created_at                           :datetime         not null
 #  updated_at                           :datetime         not null
+#  first_question_id                    :integer
```
Comment (chrisjluc) [inline app/models/survey.rb]: Nit: Can we move this up maybe 2nd or 3rd from the top, because it's more important than `created_at` and `updated_at`

File: app/models/survey.rb [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
 #
 
 class Survey < ActiveRecord::Base
   has_many :question_orders
   has_many :survey_responses
 
-  validates :description, :name, presence: true
+  belongs_to :first_question, foreign_key: :first_question_id, class_name: 'Question'
+
+  validates :description, :name, :first_question_id, presence: true
```
Comment (chrisjluc) [inline app/models/survey.rb]: I don't think a `first_question_id` needs to be present for a valid survey.

Example scenarios:
- User creates survey, doesn't add first question
- User creates survey with survey title and description, but doesn't add first question

They should be able to save their surveys in the above scenarios.

File: spec/models/survey_spec.rb [hunk 3]
```diff
@@ -10,4 +10,9 @@
     subject { FactoryGirl.build(:survey, name: "") }
     it { is_expected.not_to be_valid }
   end
+
+  context "survey is missing a first question" do
```
Comment (chrisjluc) [inline spec/models/survey_spec.rb]: Need to change this test, if we all first_question to be empty

File: app/models/survey.rb (hunk 1 above)
Comment (katrinacrisostomo) [reply app/models/survey.rb]: 👍 good point

File: app/models/survey.rb (hunk 2 above)
Comment (katrinacrisostomo) [reply app/models/survey.rb]: okaaay true, I was thinking we wouldn't allow them to create a survey without any questions but I guess why not

File: spec/models/survey_spec.rb (hunk 3 above)
Comment (katrinacrisostomo) [reply spec/models/survey_spec.rb]: I guess I would just delete this test altogether!

**Comment by katrinacrisostomo** [comment]
Addressed comments, ready for another pass of code review 👍

File: spec/factories/surveys.rb [hunk 4]
```diff
@@ -2,5 +2,6 @@
   factory :survey do
     name 'hello world'
     description 'a test survey'
+    first_question_id 1
```
Comment (chrisjluc) [inline spec/factories/surveys.rb]: Why is this set to 1?

I think it's better to replace this line with `first_question` with nothing after because I think the factory will create a new real record automagically. While, what you have it will set it to 1, but there might not be a question with id 1.

Comment (katrinacrisostomo) [reply spec/factories/surveys.rb]: oh ya my bad, forgot to delete with the spec

**Comment by chrisjluc** [comment]
Hmmm might be better to have `first_question` by default, because that seems like the more common test case rather than survey without first question, instead of just deleting it.

**Comment by katrinacrisostomo** [comment]
Okaaay done

**Comment by chrisjluc** [comment]
Can you write a test to check that is valid and that it's a question model?

Similar to `question_spec.rb`

**Comment by chrisjluc** [comment]
LGTM
